CMakeLists.txt

```diff
@@ -11,40 +11,51 @@ if (NOT CFE_SYSTEM_PSPNAME)
 endif()
 
 set(CFE_PSP_TARGETNAME "${CFE_SYSTEM_PSPNAME}")
-add_definitions(-D_CFE_PSP_)
+add_definitions(-D_CFE_PSP_)       # macro to indicate PSP scope
+
+# The "psp_module_api" defines the interface between internal PSP components
+add_library(psp_module_api INTERFACE)
+target_compile_definitions(psp_module_api INTERFACE
+    $<TARGET_PROPERTY:osal,INTERFACE_COMPILE_DEFINITIONS> # use defs from OSAL
+)
+target_include_directories(psp_module_api INTERFACE
+    fsw/inc         # public API
+    fsw/shared/inc  # all PSP shared headers
+    ${CFE_SOURCE_DIR}/cmake/target/inc    # for sysconfig
+    $<TARGET_PROPERTY:osal,INTERFACE_INCLUDE_DIRECTORIES>  # use headers from OSAL
+)
+
 
 # The PSP is currently built in two parts, consisting of a fully platform-specific
 # module combined with a shared component which is built for multiple targets.
 # The "shared" component is compiled using headers from the platform-specific module
 # so it is still ultimately a platform-specific binary, and it all gets wrapped into
 # a single PSP static library target.
-include_directories(
-    fsw/inc
-    fsw/shared/inc  # all local stuff
-    ${CFE_SOURCE_DIR}/cmake/target/inc    # for sysconfig
-    $<TARGET_PROPERTY:osal,INTERFACE_INCLUDE_DIRECTORIES>  # headers from OSAL
-)
 
 add_subdirectory(fsw/${CFE_PSP_TARGETNAME} ${CFE_PSP_TARGETNAME}-impl)
-target_compile_definitions(psp-${CFE_PSP_TARGETNAME}-impl PUBLIC
-    $<TARGET_PROPERTY:osal,INTERFACE_COMPILE_DEFINITIONS> # defs from OSAL
-)
+#target_compile_definitions(psp-${CFE_PSP_TARGETNAME}-impl PUBLIC
+#    $<TARGET_PROPERTY:psp_module_api,INTERFACE_COMPILE_DEFINITIONS>
+#)
+#target_include_directories(psp-${CFE_PSP_TARGETNAME}-impl PRIVATE
+#    $<TARGET_PROPERTY:psp_module_api,INTERFACE_INCLUDE_DIRECTORIES>
+#)
 
 add_subdirectory(fsw/shared ${CFE_PSP_TARGETNAME}-shared)
-target_compile_definitions(psp-${CFE_PSP_TARGETNAME}-shared PUBLIC
-    $<TARGET_PROPERTY:osal,INTERFACE_COMPILE_DEFINITIONS> # defs from OSAL
-)
+#target_compile_definitions(psp-${CFE_PSP_TARGETNAME}-shared PRIVATE
+#    $<TARGET_PROPERTY:psp_module_api,INTERFACE_COMPILE_DEFINITIONS>
+#)
+#target_include_directories(psp-${CFE_PSP_TARGETNAME}-shared PRIVATE
+#    $<TARGET_PROPERTY:psp_module_api,INTERFACE_INCLUDE_DIRECTORIES>
+#)
```

Comment:
Comment cleanup

Reply:
thanks for the catch, corrected in 147ac40